### PR TITLE
[MMCA-4440] - Remove unused code and ToDos through out the services 

### PR DESCRIPTION
**Pull request by tyagikumarvineet**
Description - Code changes to make the repo code ready to hand it over to live

Below have been covered as part of this PR

- remove unused code
- remove comments
- add scala style config and plugin
- remove warnings
- remove magic numbers
- update it test settings to the latest one 
- refactoring

### CODE REVIEW

File: scalastyle-config.xml [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    <check level="warning" class="org.scalastyle.file.NoNewLineAtEofChecker" enabled="false"/>
+    <check  level="warning" class="org.scalastyle.scalariform.MultipleStringLiteralsChecker" enabled="false">
+        <parameters>
+            <parameter name="allowed">1</parameter>
+            <parameter name="ignoreRegex">^\&quot;\&quot;$</parameter>
+        </parameters>
+    </check>
+    <check enabled="true" class="org.scalastyle.scalariform.NoCloneChecker" level="warning"/>
+    <check enabled="true" class="org.scalastyle.scalariform.NotImplementedErrorUsage" level="warning"/>
+</scalastyle>
```
Comment (AdamCoops) [inline scalastyle-config.xml:126]: Can you add a empty line at the end to remove this end of file error?
It's weird that it adds these to a .xml file

File: app/models/repositories/EmailRepositoryResult.scala [hunk 2]
```diff
@@ -17,6 +17,7 @@
 package models.repositories
 
 sealed trait EmailRepositoryResult
+
 case object SuccessfulEmail extends EmailRepositoryResult
+
 case object FailedToRetrieveEmail extends EmailRepositoryResult
```
Comment (AdamCoops) [inline app/models/repositories/EmailRepositoryResult.scala:23]: Can we add a new to the end of this file? it looks like it doesn't have to me. ignore this if its correct

File: build.sbt [hunk 3]
```diff
@@ -19,15 +24,23 @@ lazy val microservice = Project(appName, file("."))
...
-  .settings(integrationTestSettings() *)
   .settings(resolvers += Resolver.jcenterRepo)
   .settings(Test / parallelExecution := false)
   .settings(addTestReportOption(IntegrationTest, "int-test-reports"))
   .disablePlugins(sbt.plugins.JUnitXmlReportPlugin)
 
-val bootstrap = "7.22.0"
+lazy val it = project
+  .enablePlugins(PlayScala)
+  .dependsOn(microservice % "test->test") // the "test->test" allows reusing test code and test dependencies
```
Comment (AdamCoops) [inline build.sbt]: I think the story wants comments like this removed

File: scalastyle-config.xml [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    <check level="warning" class="org.scalastyle.file.FileTabChecker" enabled="true"/>
+    <check level="warning" class="org.scalastyle.file.FileLengthChecker" enabled="true">
+        <parameters>
+            <parameter name="maxFileLength"><![CDATA[800]]></parameter>
+        </parameters>
+    </check>
+    <check level="warning" class="org.scalastyle.file.HeaderMatchesChecker" enabled="false">
+        <parameters>
+            <parameter name="header"><![CDATA[// Copyright (C) 2011-2012 the original author or authors.
+// See the LICENCE.txt file distributed with this work for additional
```
Comment (AdamCoops) [inline scalastyle-config.xml]: is this big comment correct? Can we reformat this so it matches the indentation of the text assuming it's suppose to be here.

File: test/config/AppConfigSpec.scala [hunk 5]
```diff
@@ -16,44 +16,102 @@
...
-    val configuration = Configuration.load(env)
-    val servicesConfig = new ServicesConfig(configuration)
-  }
+  "url" should {
+
+    "remove left hand side slash" in new Setup {
+
+      import appConfig.URLSyntacticSugar
+
+      val url: String = "http://localhost/" / "abcd"
```
Comment (AdamCoops) [inline test/config/AppConfigSpec.scala]: both localhost and abcd can be moved into Setup if you're wanting to reduce string literals.

File: test/models/MdgSub09ResponseSpec.scala [hunk 6]
```diff
@@ -19,35 +19,42 @@ package models
...
 
     "parse message no email and no timestamp" in {
-      val sub09Response = Sub09Response.noEmailNoTimestamp(EORI1)
+      val sub09Response = Sub09Response.noEmailNoTimestamp(eori1)
+
       val result = MdgSub09Response.sub09Reads.reads(sub09Response).get
+
       result mustBe MdgSub09Response(None, None)
     }
   }
```
Comment (AdamCoops) [inline test/models/MdgSub09ResponseSpec.scala:55]: I think you can delete this space

File: app/models/repositories/EmailRepositoryResult.scala (hunk 2 above)
Comment (tyagikumarvineet) [reply app/models/repositories/EmailRepositoryResult.scala:23]: It already has a line break at the end

File: test/models/MdgSub09ResponseSpec.scala (hunk 6 above)
Comment (tyagikumarvineet) [reply test/models/MdgSub09ResponseSpec.scala:55]: Adam, i have given this line break to differentiate between val and the assertion

File: build.sbt (hunk 3 above)
Comment (tyagikumarvineet) [reply build.sbt]: Removed.

File: scalastyle-config.xml (hunk 1 above)
Comment (tyagikumarvineet) [reply scalastyle-config.xml:126]: New line break added.

File: scalastyle-config.xml (hunk 4 above)
Comment (tyagikumarvineet) [reply scalastyle-config.xml]: Yes Adam, it needs to be there as it includes the mandatory licensing advisory. 

Done

**Review by AdamCoops — APPROVED** [review]

**Pull request merged**